daemon/buildinfo.c

```diff
@@ -200,8 +200,67 @@
 
 #define FEAT_YES_NO(x) ((x) ? "YES" : "NO")
 
+// coverity[ +tainted_string_sanitize_content : arg-0 ]
+static inline void coverity_remove_taint(char *s)
+{
+    (void)s;
+}
+
+void read_install_type(void) {
+    char *filepath;
+    filepath = mallocz(sizeof(char) * (strlen(netdata_configured_user_config_dir) + strlen(".install-type") + 2));
+    sprintf(filepath, "%s/%s", netdata_configured_user_config_dir, ".install-type");
+    FILE *fp = fopen(filepath, "r");
+
+    if(fp) {
+        char *line = mallocz(sizeof(char) * (200 + 1));
+        while (fgets(line, 200, fp) != NULL) {
+            char *value = line;
+
+            while (*value && *value != '=') value++;
+
+            if (*value == '=') {
+                *value = '\0';
+                value++;
+                if (*value == '\'' || *value == '\"') value++;
+                char *end = value;
+                while (*end && *end != '\'' && *end != '\"' && *end != '\n') end++;
+                *end = '\0';
+                coverity_remove_taint(line);
+                coverity_remove_taint(value);
+            } else {
+                continue;
+            }
+
+            setenv(line, value, 1);
+        }
+        fclose(fp);
+        freez(line);
+    } else {
+        setenv("INSTALL_TYPE", "unknown", 1);
```

Comment:
isnt this handled already in `print_build_info` by:
```
    if (install_type == NULL) {
        install_type = "unknown";
    }
```

You handle it there in case where `fgets` kicks you out in case you dont find `=` anyway

Reply:
Good point.

I’m far too used to having to account for implicit exit paths from functions resulting from exceptions (which, of course, are a non-issue in plain C).